analyses/independent-samples-release/util/independent-rna-samples.R

```diff
@@ -0,0 +1,115 @@
+  matched_rna <- histology_df %>%
+    # keep rna from histology_df
+    dplyr::filter(experimental_strategy == "RNA-Seq",
+                  # find participants which have matching dna samples in independent_wgswxs
+                  Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID,
+                  # keep specific sample_ids since some participants might have multiple sample_ids
+                  sample_id %in% independent_dna$sample_id)
+  
+  # has rna samples which match the independent samples provided
+  sample_df <- matched_rna
+
+  
+  # Here we are adding only initial only-RNA-Seq samples
+  # since this will always to part of independent_dna_plus_only_rna
+  # regardless tumor_description_rna_only is "primary" OR "primary_plus"
+  #
+  if( match_type == "independent_dna_plus_only_rna" ) {
+    # find sample set where we initial only-RNA-Seq samples
+    only_rna_intial <- histology_df %>% 
+      # keep rna from histology_df
+      dplyr::filter(experimental_strategy == "RNA-Seq",
+                    tumor_descriptor %in% primary_descs,
+                    # find and remove participants which have 
+                    # matching dna samples in independent_wgswxs 
+                    !Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID)
+    
+    # has rna samples which match the independent samples provided plus rna only sample which are primary tumors
+    sample_df <- bind_rows(sample_df,only_rna_intial)
+  }
+  
+  
+  # Here we are adding only-RNA-Seq samples which are not initial
+  # if tumor_description_rna_only == "primary_plus"
+  #
+  if(match_type == "independent_dna_plus_only_rna" & tumor_description_rna_only == "primary_plus"){
+    # find sample set where we only find rna samples
+    only_rna_plus <- histology_df %>% 
+      # keep rna from histology_df
+      dplyr::filter(experimental_strategy == "RNA-Seq",
+                    # find and remove participants which have 
+                    # matching dna samples in independend_wgswxs 
+                    !Kids_First_Participant_ID %in% independent_dna$Kids_First_Participant_ID,
+                    # and participant not in only_rna_initial sample set
+                    !Kids_First_Participant_ID %in% only_rna_intial$Kids_First_Participant_ID
+                    )
+    # has rna samples which match the independent samples provided plus rna only sample which are primary tumors plus rna samples where no primary primaries exists
+    sample_df <- bind_rows(sample_df,only_rna_plus)
+  } 
+  
+  # get the samples from the earliest timepoints for each Participant
+  # age_at_diagnosis_days is no longer relevant,
+  # as it is the same for all samples from an participant, but
+  # leaving this in for future use in case we get specimen order data 
+  early_samples <- sample_df %>%
+    dplyr::group_by(Kids_First_Participant_ID) %>%
+    dplyr::summarize(age_at_diagnosis_days = min(age_at_diagnosis_days)) %>%
```

Comment:
```suggestion
    dplyr::summarize(age_at_diagnosis_days = min(as.numeric(age_at_diagnosis_days))) %>%
```

`age_at_diagnosis_days` may be read as `character` type by `col_guess`. If `age_at_diagnosis_days` is `character` type:

```text
> min(c("1980", "980"))
[1] "1980"
```

`analyses/independent-samples-release/util/independent-dna-samples.R` converts `age_at_diagnosis_days` to numeric type with

```R
  sample_df <- sample_df %>%
    dplyr::mutate(age_at_diagnosis_days = as.numeric(age_at_diagnosis_days))
```

`age_at_diagnosis_days` is read as `character` type in docker using code from `analyses/independent-samples-release/02-generate-independent-rnaseq.R`:

```text
> readr::read_tsv("data/v11/histologies-base.tsv", 
+                 guess_max = 10000,
+                 col_types = readr::cols())[, "age_at_diagnosis_days"]
...
# A tibble: 39,892 x 1
   age_at_diagnosis_days
   <chr>                
 1 1872                 
 2 NA                   
 3 980                  
 4 4938                 
 5 4738                 
 6 4754                 
 7 1546                 
 8 1569                 
 9 NA                   
10 3316                 
# … with 39,882 more rows
```

Reply:
updated this on file read in instead of in functions